### ISSUE TITLE
[wg/data-shapes] Data Shapes WG Charter

### ISSUE THREAD

**Issue by nicholascar** [w3c/strategy#476]
New charter proposal, reviewers please take note.

# Charter Review

[Proposed Data Shapes WG Charter](https://w3c.github.io/charter-drafts/2024/data-shapes.html)
[diff from charter template](https://services.w3.org/htmldiff?doc1=https://w3c.github.io/charter-drafts/charter-template.html&doc2=https://w3c.github.io/charter-drafts/2024/data-shapes.html)

What kind of charter is this? Check the relevant box / remove irrelevant branches.

* New

- [x] New WG
- [ ] New IG

SHACL was originally produced by the [RDF Data Shapes Working Group](https://www.w3.org/groups/wg/data-shapes/).

If this is a new WG or IG charter request, link to [Advance Notice](https://github.com/w3c/strategy/issues/476), and any issue discussion.

Horizontal Reviews: apply the Github label "Horizontal review requested" to request reviews for accessibility (a11y), internationalization (i18n), privacy, security, and TAG. **Also add a "card" for this issue to the [Strategy Funnel](https://github.com/w3c/strategy/projects/2)**.

Communities suggested for outreach: [RDF-Star WG](https://www.w3.org/groups/wg/rdf-star/)

Known or potential areas of concern: _none_

Where would charter proponents like to see issues raised? 
[in charter-drafts GH](https://github.com/w3c/charter-drafts/issues/) or [This issue](https://github.com/w3c/strategy/issues/476)

Anything else we should think about as we review? _not that we know of_

Note: proposed chairs should be copied @... on this issue.

@nicholascar @PapoutsoglouE

**Comment by plehegar** [comment]
Any reason why we can't reuse the original name of the Working Group? [RDF Data Shapes Working Group](https://www.w3.org/groups/wg/data-shapes/)

**Comment by caribouW3** [comment]
The shortname was already 'data shapes', so it seems to make more sense.

**Comment by plehegar** [comment]
[Advance notice](https://lists.w3.org/Archives/Member/w3c-ac-members/2024JulSep/0047.html)

**Comment by himorin** [comment]
correct link for diff from template: https://services.w3.org/htmldiff?doc1=https://w3c.github.io/charter-drafts/charter-template.html&doc2=https://w3c.github.io/charter-drafts/2024/data-shapes.html

**Comment by himorin** [comment]
(not within i18n review)

- In 3.2 non-normative documents contains a text `The group may produce other Community Group Reports`, but there seems no mention about [SHACL Community Group](https://www.w3.org/groups/cg/shacl/)
- links to [SHACL 1.2 drafts](https://w3c.github.io/shacl/) are failing to have charter-drafts repository as its base (not sure why)

**Comment by plehegar** [comment]
As a side, PING got a few suggestions (from @pes10k, not privacy related):
* copy-editing improvements:
  * the only sentence in the 3 paragraph Motivation and Background section thats doing any work / needed is “The overarching goal of the SDWWG is to enable spatial data to be integrated within the wider Web of data; developing and maintaining standard patterns and specifications addressing these problems.” (not a normative concern, of course)
  * Again, there are abbreviations and references to other specs and technologies all over this charter that are not clear to the reader. I have no idea what a “SSN Ontology” or “SOSA vocabulary” vocabulary are when they’re described in the Scope section.  Just some links and things would help!
* describe relationship to the previous WG in the background

**Comment by caribouW3** [comment]
> * describe relationship to the previous WG in the background

I added this text to refer to the previous group name: "This group will maintain specifications developed by the former RDF Data Shapes WG."

**Comment by himorin** [comment]
no comment or request from i18n

**Comment by simoneonofri** [comment]
From a security perspective, it's useful to remember to include in Security Considerations, the possibility of attacks such as SPARQL Injection.

**Comment by caribouW3** [comment]
> From a security perspective, it's useful to remember to include in Security Considerations, the possibility of attacks such as SPARQL Injection.

Good point. It seems to me that it ought to filed as a spec issue (the security consideration in https://github.com/w3c/shacl/tree/main/shacl-sparql is short and very generic) rather than a charter issue. Do you want to file an issue there?

**Comment by nicholascar** [comment]
> > From a security perspective, it's useful to remember to include in Security Considerations, the possibility of attacks such as SPARQL Injection.
> 
> Good point. It seems to me that it ought to filed as a spec issue (the security consideration in https://github.com/w3c/shacl/tree/main/shacl-sparql is short and very generic) rather than a charter issue. Do you want to file an issue there?

Yes, we see security as applicable per-Spec, but it;s hard to think of generic yet still useful things to say in the Charter.

**Comment by ruoxiran** [comment]
no comment from APA.

**Comment by plehegar** [comment]
section 1: the background says "This group will maintain specifications developed by the former RDF Data Shapes WG.", which seems to be a statement about the scope of the Group more than the background. Maybe that sentence should move into the in-scope section?

section 2.1: out of scope section seems awkward: in scope makes it clear that the Group will develop specifications based on SHACL. Why the need to exclude something which is not include in the in-scope section?

section 3.1: for SHACL 1.2 Compact Syntax, it says "Status: Draft (from previous WG work)". The draft seems to be from the CG, not the previous WG.

section 5.1: this needs an update to describe the relation between RDF-Star and the Data Shapes Working Group. In other words, why is RDF-Start WG listed there?

section 5.2: s/None yet identified./None identified./

section 11.1: the table needs to be reduced to the "Initial Charter" entry, since this is not a rechartering.

**Comment by caribouW3** [comment]
> section 1: the background says "This group will maintain specifications developed by the former RDF Data Shapes WG.", which seems to be a statement about the scope of the Group more than the background. Maybe that sentence should move into the in-scope section?

This was added as a resolution of [this comment](https://github.com/w3c/strategy/issues/476#issuecomment-2359328546), which was asking about background and previous group specifically. 

> section 2.1: out of scope section seems awkward: in scope makes it clear that the Group will develop specifications based on SHACL. Why the need to exclude something which is not include in the in-scope section?

IMHO the purpose of an out-of-scope section is to clarify that some other work that could also apply to the same use cases, scenarios, and/or solve similar or connected issues are excluded from the charter. 
The charter template says it includes an out-of-scope section also for legal purposes (IPR commitments, most likely).
 
> section 3.1: for SHACL 1.2 Compact Syntax, it says "Status: Draft (from previous WG work)". The draft seems to be from the CG, not the previous WG.

I removed "(from previous WG work)"  since there's apparently no published draft from the RDF Data Shapes WG.
 
> section 5.1: this needs an update to describe the relation between RDF-Star and the Data Shapes Working Group. In other words, why is RDF-Start WG listed there?

SHACL being completely RDF-based, I don't think it's needed to re-state that there's a technical dependency between those specs. Do you mean something else?
 
> section 5.2: s/None yet identified./None identified./

Done.
 
> section 11.1: the table needs to be reduced to the "Initial Charter" entry, since this is not a rechartering.

Done.

**Comment by svgeesus** [comment]
In Success Criteria, why has template wording on security, privacy and accessibility been removed?

Same question for TAG Web Platform Design Principles .

**Comment by caribouW3** [comment]
> In Success Criteria, why has template wording on security, privacy and accessibility been removed?
> 
> Same question for TAG Web Platform Design Principles .

The template says it is  "For specifications of technologies that directly impact user experience, such as content technologies, as well as protocols and APIs which impact content", which is not the case here. Those specs are not about Web design either, so if you check the design principles, it's unlikely that anything applies.

That said, Horizontal review by Accessibility/Security/Privacy is of course cited in the Coordination section, which also has a sentence about the TAG review.

**Comment by svgeesus** [comment]
Thanks for the explanation, agreed

**Comment by caribouW3** [comment]
[AC Review started](https://lists.w3.org/Archives/Member/w3c-ac-members/2024OctDec/0032.html)

**Comment by VladimirAlexiev** [comment]
A couple of comments
- where is Holger Knublauch on this?
- knowing how difficult and bureaucratic  WG proceedings are, is there an easier way? Guess not 
- maybe also include in scope:  SHACL and Graphs (eg see https://github.com/Sveino/Inst4CIM-KG/issues/140, not very well written...)

**Comment by HolgerKnublauch** [comment]
> * where is @HolgerKnublauch on this? There'll be no competition/bickering between the existing CG and the proposed WG, right?

I am fully supportive and have been in close contact with @nicholascar from the beginning. We (TopQuadrant) has started the W3C membership process and I'd be available to continue as editor if the group wants me to.

**Comment by nicholascar** [comment]
@VladimirAlexiev yes, we are supporting of @afs's **Extending SHACL to RDF Datasets** and my engineers working on SHACL pipelines right now are using SHACL with multi-graph hacks right now, so we are keen on moving things forward here. 

I assume, @HolgerKnublauch to confirm, that multigraph/dataset extensions for SHACL will appear in the SHACL 1.2 Core and SPARQL docs as extensions to the section 3.3 already there, so don't need a new WG deliverable. 

If this is correct then @VladimirAlexiev we'll just need your (and @afs's) input on the eventual scope of those docs, which are of course, already drafted.

I'm interested in packaging SHACL shapes in named, data objects (currently most people use an `owl:Ontology`) so we can implement distributed inheritance hierarchies, e.g. Shapes for SKOS & dependent, extended Shapes for profiles of SKOS... So this is the expected target of SHAP Profiling, as opposed to SHACL 1.2 Core or SHACL 1.2 SPARQL.

**Comment by plehegar** [comment]
[Announced](https://lists.w3.org/Archives/Member/w3c-ac-members/2024OctDec/0076.html)
